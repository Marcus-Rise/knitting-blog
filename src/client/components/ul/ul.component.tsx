import React from "react";
import styles from "./ul.module.scss";

interface IProps {
  items: string[];
}

const Ul: React.FC<IProps> = (props) => {
  const items = props.items.map((i, index) => (
    <li className={styles.li} key={index}>
      {i}
    </li>
  ));

  return <ul>{items}</ul>;
};

export { Ul };
