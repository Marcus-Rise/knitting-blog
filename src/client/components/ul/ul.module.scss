.li {
  line-height: 1.75rem;
  letter-spacing: .2em;

  &::marker {
    color: var(--accent);
    font-size: 1.3rem;
  }
}
