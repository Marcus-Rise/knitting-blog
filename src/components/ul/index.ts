export { Ul } from "./ul.component";
