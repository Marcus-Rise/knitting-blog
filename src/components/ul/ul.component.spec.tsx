import { render } from "@testing-library/react";
import { Ul } from "./ul.component";
import React from "react";

describe("Ul", () => {
  test("render", () => {
    const { asFragment } = render(<Ul items={["item", "item2"]} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
