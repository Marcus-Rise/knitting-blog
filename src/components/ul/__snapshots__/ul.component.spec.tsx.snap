// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ul render 1`] = `
<DocumentFragment>
  <ul>
    <li
      class="li"
    >
      item
    </li>
    <li
      class="li"
    >
      item2
    </li>
  </ul>
</DocumentFragment>
`;
